File: javascript/leetcode/231. Power of Two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function (n) {
  if (n > 0) {
    while (n % 2 == 0 && n > 2) n >>= 1;
  }

  return n == 1 || n == 2;
};

File: c++/leetcode/231. Power of Two.cpp
class Solution
{
public:
  bool isPowerOfTwo(int n)
  {

    if (n > 0)
    {
      while (n > 2 && n % 2 == 0)
        n >>= 1;
    }

    return n == 1 || n == 2;
  }
};
